test/Concurrency/Runtime/async_sequence.swift

```diff
@@ -0,0 +1,524 @@
+  expectEqual(results, [])
+}
+
+AsyncMapSequenceTests.test("map throwing") {
+  let seq = [1, 2, 3].async
+    .map { value throws -> String in
+      if value == 2 {
+        throw Failure()
+      }
+      return "\(value)" 
+    }
+  var it = seq.makeAsyncIterator()
+  var results = [String]()
+  do {
+    while let value = try await it.next() {
+      results.append(value)
+    }
+    expectUnreachable()
+  } catch {
+    expectEqual(results, ["1"])
+  }
+}
+
+var AsyncFilterSequenceTests = TestSuite("AsyncFilterSequence")
+
+AsyncFilterSequenceTests.test("filter out one item") {
+  let results = await [1, 2, 3].async.filter { $0 % 2 != 0 } .collect()
+  expectEqual(results, [1, 3])
+}
+
+AsyncFilterSequenceTests.test("filter out one item throwing") {
+  do {
+    let results = try await [1, 2, 3].async
+      .filter { try throwing($0 % 2 != 0) }
+      .collect()
+    expectEqual(results, [1, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncFilterSequenceTests.test("filter out one item throwing async") {
+  do {
+    let results = try await [1, 2, 3].async
+      .filter { try await asynchronousThrowing($0 % 2 != 0) }
+      .collect()
+    expectEqual(results, [1, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncFilterSequenceTests.test("filter out two items") {
+  let results = await [1, 2, 3].async.filter { $0 % 2 == 0 } .collect()
+  expectEqual(results, [2])
+}
+
+AsyncFilterSequenceTests.test("filter out all items") {
+  let results = await [1, 2, 3].async.filter { _ in return false } .collect()
+  expectEqual(results, [Int]())
+}
+
+AsyncFilterSequenceTests.test("filter out no items") {
+  let results = await [1, 2, 3].async.filter { _ in return true } .collect()
+  expectEqual(results, [1, 2, 3])
+}
+
+var AsyncDropWhileSequenceTests = TestSuite("AsyncDropWhileSequence")
+
+AsyncDropWhileSequenceTests.test("drop one") {
+  let results = await [1, 2, 3].async.drop { value -> Bool in
+    return value == 1
+  }.collect()
+  expectEqual(results, [2, 3])
+}
+
+AsyncDropWhileSequenceTests.test("drop one async") {
+  let results = await [1, 2, 3].async.drop { value async -> Bool in
+    return await asynchronous(value == 1)
+  }.collect()
+  expectEqual(results, [2, 3])
+}
+
+AsyncDropWhileSequenceTests.test("drop one throws") {
+  do {
+    let results = try await [1, 2, 3].async.drop { value throws -> Bool in
+      return try throwing(value == 1)
+    }.collect()
+    expectEqual(results, [2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncDropWhileSequenceTests.test("drop one throws async") {
+  do {
+    let results = try await [1, 2, 3].async.drop { value async throws -> Bool in
+      return try await asynchronousThrowing(value == 1)
+    }.collect()
+    expectEqual(results, [2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncDropWhileSequenceTests.test("drop none") {
+  let results = await [1, 2, 3].async.drop { _ in return false }.collect()
```

Comment:
The `collect()` call in all these makes me think we need an `elementsEqual` method for two async sequences.

Reply:
it could perhaps work like AsyncZip2Sequence (which is not part of this yet...)